modules/busco/main.nf

```diff
@@ -0,0 +1,48 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process BUSCO {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::busco=5.2.2" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/busco:5.2.2--pyhdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/busco:5.2.2--pyhdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(fasta)
+    path(augustus_config)
```

Comment:
I am missing handling of a database such as in the [nf-core/mag local module](https://github.com/nf-core/mag/blob/d4f00bc2d178c1c8dc25b576c9e77e70511d5d21/modules/local/busco.nf#L23-L24)

Reply:
So in this module, we give database as one of the arguments rather than an input. In tests, it works with both local database and downloaded database. This makes it more generic, rather than absolutely needing a database.

Reply:
I agree with @d4straub that it is important to support offline use. I just submitted a [pull request for a minimised lineage dataset that can be used to test the module in offline mode](https://github.com/nf-core/test-datasets/pull/392). The offline lineage datasets can be passed to the module via an extra channel, with the possibility to keep it empty like for `path(augustus_config)`.  I was using something like below in some local modules.

```
if (lineage) options.args += "--offline --lineage_dataset $lineage"
```

Shall we follow this approach, using `--download_path` instead?

Reply:
> So in this module, we give database as one of the arguments rather than an input. In tests, it works with both local database and downloaded database. This makes it more generic, rather than absolutely needing a database.

Where? Which arguments? Do you mean `option.args`? If yes, than thats possible but this is typically solved by a separate input channel. E.g.:
```suggestion
    input:
    tuple val(meta), path(fasta)
    path(augustus_config)
    path(database_path)
```
and using than in the `script:` block
`    def database = database_path ? "--download_path $database_path" : ""`
And than use 
```
    busco \\
        $options.args \\
        --augustus \\
        $database \\
        --cpu $task.cpus \\
        --in  $fasta \\
        --out $meta.id
```

> Shall we follow this approach, using --download_path instead?

I think so, yes, because it is more flexible.

Reply:
all input files, including databases, should be passed as an input channel - only like that can nextflow take care of mounting files into containers, upload them to AWS worker nodes etc.